lib/srv/db/mongodb/protocol/opreply.go

```diff
@@ -138,6 +138,26 @@ func readOpReply(header MessageHeader, payload []byte) (*MessageOpReply, error)
 	}, nil
 }
 
+// ReadReplyDocuments reads multiple documents from the source.
+//
+// This function works in the same way as wiremessage.ReadReplyDocuments except, it can handle document of size 0.
+// When a document of size 0 is passed to wiremessage.ReadReplyDocuments, it will keep creating empty documents until
+// it uses all system memory/application crash.
+func ReadReplyDocuments(src []byte) (docs []bsoncore.Document, rem []byte, ok bool) {
+	rem = src
+	for {
+		var doc bsoncore.Document
+		doc, rem, ok = bsoncore.ReadDocument(rem)
+		if !ok || len(doc) == 0 {
```

Comment:
It looks reasonable but I think that we also need to  raise an issue in MongoDB driver lib to get this fix in the origin -  I will do that.